src/validation/utils.js

```diff
@@ -0,0 +1,87 @@
+import { ERROR_MESSAGES as Errors, EMPTY_VALUE, TYPES } from './constants';
+
+/**
+ * Check value type.
+ * @param {*} value - The value to check.
+ * @param {function} callback - The typechecking function.
+ * @throws {TypeError} When value is not of the expected type.
+ *
+ * @example
+ * validateType(5, isBoolean); // Error: Value must be of type boolean.
+ */
+export function validateType(value, callback) {
```

Comment:
If these utils are intended to be used by developers rather than requiring a user to pass the callback themselves I think there could be a better way to handle it. But that would take some thought. If intended to be a private function then this should be fine

Reply:
I didn't want to have a separate function for each type that's why I opted for the callback. It's not really meant to be exported outside of the project.